### PR TITLE
Update/october 2021

### PR DESCRIPTION
**Pull request by pauloxnet**

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -59,4 +59,4 @@ repos:
     rev: "1.7.0"
     hooks:
       - id: bandit
-        args: ["--exclude", ".venv,tests"]
+        args: ["--exclude", ".venv,tests", "--skip", "B108"]
```
Comment (trottomv) [inline .pre-commit-config.yaml:62]: I would not turn on the overall sikipping of the B108 error, from what I understand it is linked to the failed security check "hardcoded temp dir" of line 24 of gunicorn.conf.py.
I would personally remove `--skip B108` from here, suggesting alternative workarounds:
- add a `# nosec` comment at the end of row 24 of `gunicorn.conf.py`
- set the temporary directory as environment variable `os.getenv("WORKER_TMP_DIR", tempfile.gettempdir())` without an hardcoded fallback, but using tempfile as suggested here https://security.openstack.org/guidelines/dg_using-temporary-files-securely.html

File: {{cookiecutter.project_dirname}}/.pre-commit-config.yaml [hunk 2]
```diff
@@ -66,4 +66,4 @@ repos:
     rev: "1.7.0"
     hooks:
       - id: bandit
-        args: ["--exclude", ".venv,tests"]
+        args: ["--exclude", ".venv,tests", "--skip", "B108"]
```
Comment (trottomv) [inline {{cookiecutter.project_dirname}}/.pre-commit-config.yaml:69]: like above

File: {{cookiecutter.project_dirname}}/gunicorn.conf.py [hunk 3]
```diff
@@ -17,3 +17,8 @@
 # https://docs.gunicorn.org/en/stable/settings.html#worker-processes
 
 worker_class = "uvicorn.workers.UvicornWorker"
+
+# Temporary Directory
+# https://docs.gunicorn.org/en/stable/settings.html#worker-tmp-dir
+
+worker_tmp_dir = "/dev/shm"
```
Comment (trottomv) [inline {{cookiecutter.project_dirname}}/gunicorn.conf.py:24]: see comment on `--skip B108`

File: {{cookiecutter.project_dirname}}/scripts/check.sh [hunk 4]
```diff
@@ -1,10 +1,10 @@
...
-set -ex
+set -euo pipefail
 
 python3 -m black --check .
 python3 -m isort --check .
 python3 -m flake8
 python3 -m mypy .
-python3 -m bandit -r -q -x tests .
-python3 -m jake ddt -q -r requirements/remote.txt
+python3 -m bandit --quiet --recursive --skip B108 --exclude tests .
```
Comment (trottomv) [inline {{cookiecutter.project_dirname}}/scripts/check.sh:9]: linked on comment above on `--skip B108`

File: {{cookiecutter.project_dirname}}/Dockerfile [hunk 5]
```diff
@@ -6,58 +6,66 @@ ENV APPUSER=$USER LANG=C.UTF-8 LC_ALL=C.UTF-8 PYTHONUNBUFFERED=1 PYTHONDONTWRITE
...
-RUN python3 -m pip install --no-cache-dir -r requirements/remote.txt
+RUN python3 -m pip install --user --no-cache-dir -r requirements/remote.txt \
+    && find ${PACKAGES_PATH}/botocore/data/* -not -name s3 -delete \
+    && find ${PACKAGES_PATH}/babel/locale-data/* -not -name en* -not -name it* -not -name root* -delete
 COPY --chown=$APPUSER . .
-USER $APPUSER
-CMD python3 manage.py migrate --noinput && \
-    python3 manage.py collectstatic --clear --noinput && \
-    gunicorn {{cookiecutter.project_slug}}.asgi:application
+ENTRYPOINT ["./scripts/entrypoint.sh"]
```
[the remark this answers is not in the document]
Comment (pauloxnet) [reply {{cookiecutter.project_dirname}}/Dockerfile:51]: Are you sure is not an issue related to docker for mac ?

[the remark this answers is not in the document]
Comment (pauloxnet) [reply {{cookiecutter.project_dirname}}/Dockerfile:51]: It seems you forgot to add execute permission to your `entrypoint.sh` in your project generated with this cookiecutter.

See here to fix:
https://stackoverflow.com/a/62839515/755343

**Pull request merged**